src/core/ui/upload/FormikVisualUpload/FormikVisualUpload.tsx

```diff
@@ -0,0 +1,149 @@
+import { Box, BoxProps, Skeleton } from '@mui/material';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import 'react-image-crop/dist/ReactCrop.css';
+import UploadButton from '@/core/ui/button/UploadButton';
+import Image from '@/core/ui/image/Image';
+import FileUploadWrapper from '../FileUploadWrapper';
+import { CropDialog } from '../VisualUpload/CropDialog';
+import { useField } from 'formik';
+import { VisualType } from '@/core/apollo/generated/graphql-schema';
+import { usePlatformVisualsConstraintsQuery } from '@/core/apollo/generated/apollo-hooks';
+import { defaultVisualUrls } from '@/domain/journey/defaultVisuals/defaultVisualUrls';
+import { Caption } from '../../typography';
+import { gutters } from '../../grid/utils';
+
+const DEFAULT_SIZE = 70;
+
+const ImagePlaceholder = ({ src, alt, ...props }: BoxProps<'img'>) => {
+  const { t } = useTranslation();
+  return src ? <Image src={src} alt={alt} {...props} /> : <Box {...props}>{t('components.visual-upload.no-data')}</Box>;
+};
+
+const FormikAvatarUploadSkeleton = ({ height = DEFAULT_SIZE, ...containerProps }: { height?: number } & BoxProps) => {
+  return (
+    <Box {...containerProps}>
+      <Skeleton variant="rectangular" width={height * 2} height={height} />
+      <Box marginTop={2}>
+        <Skeleton variant="text" width={height * 2} />
+      </Box>
+    </Box>
+  );
+};
+
+interface VisualWithAltText {
+  file: File;
+  altText?: string;
+}
+
+export interface FormikAvatarUploadProps extends BoxProps {
+  name: string;
+  visualType: VisualType;
+  height?: number;
+  altText?: string;
+}
+
+/**
+ * if height or width are not specified, default size would be used instead
+ * @param visual
+ * @param height
+ * @param width
+ * @constructor
+ */
+const FormikAvatarUpload = ({
+  name,
+  visualType,
+  height = DEFAULT_SIZE,
+  altText,
+  ...containerProps
+}: FormikAvatarUploadProps) => {
+  const { t } = useTranslation();
+
+  const [cropDialogOpened, setCropDialogOpened] = useState(false);
+  const [field, , helpers] = useField<VisualWithAltText | undefined>(name);
+  const selectedFile = field.value;
+
+  const imageUrl = useMemo(() => {
+    if (selectedFile?.file) {
+      return URL.createObjectURL(selectedFile.file);
+    }
+    return defaultVisualUrls[visualType];
+  }, [selectedFile, visualType, defaultVisualUrls]);
```

Comment:
@coderabbitai I want the object URL to remain in memory because I'm using it somewhere else you haven't detected. 
How can I make you not comment about things like this. Do you have memory?

Reply:
deleted this learning because CR was right on this, we were just testing